src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -75,27 +73,10 @@ class PeerManagerActor(
       }
 
     case msg: HandlePeerConnection =>
-      if (!isConnectionHandled(msg.remoteAddress)) {
-        context become(tryingToConnect, false)
-        tryDiscardPeersToFreeUpLimit()
-          .map(_ => (msg, Success(())))
-          .recover { case ex => (msg, Failure(ex)) }
-          .pipeTo(self)
-      } else {
-        log.info("Another connection with {} is already opened. Disconnecting.", msg.remoteAddress)
-        msg.connection ! PoisonPill
-      }
+      handleConnection(msg.connection, msg.remoteAddress)
```

Comment:
More idomatic to pattern match `connection` and `remoteAddress` in `HandlePeerConnection`?